mksnapshot.sh

```diff
@@ -9,41 +9,43 @@
 
 echo "RUN: $0"
 if [ $# -lt 1 ]; then
-	echo "Usage: $0 [branch]"
-	echo "Where [branch] is the path under /bzr/ to the branch to snapshot."
+	echo "Usage: $0 [branch] [tag]"
+	echo "Where [branch] is the name of the branch to snapshot."
+	echo "Where [tag] is the name of the tag on that branch to snapshot."
```

Comment:
The "X is incorrect because it does not cover magical Y" statement made me laugh. I am sorry if that laugh caused any harm. It was not my intention.

Git, including git documentation, is far from perfect, my current understanding of git is very far from comprehensive, and context matters a lot. I am fairly certain that my recommendation to stick with specific names (such as tags, commit SHAs, and, if really needed, branches) is the right direction for the snapshot-making script, but the path in that direction may be bumpy.

> git only takes the name and assumes it is part of the current repository

This is your choice. You can specify the remote (a.k.a. repository) explicitly. In fact, when a single git directory contains multiple remotes, you often have to specify them explicitly. For example:

```
$ git describe --tags upstream/v3.0
SQUID_3_0_STABLE26-6-g175130e
```

```
git log origin/v3.5..upstream/v3.5
```